Dockerfile

```diff
@@ -1,18 +1,31 @@
-FROM centos:7
-MAINTAINER carcassi@umich.edu
+FROM centos:centos7
 
-RUN rpm -Uvh https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
-    yum install -y yum-plugin-priorities && \
-    rpm -Uvh http://mirror.grid.uchicago.edu/pub/osg/3.4/osg-3.4-el7-release-latest.rpm
+LABEL maintainer OSG Software <help@opensciencegrid.org>
+
+# Create the squid user with a fixed GID/UID
+RUN groupadd -o -g 10941 squid
+RUN useradd -o -u 10941 -g 10941 -s /sbin/nologin -d /var/lib/squid squid
+
+
+RUN yum -y install http://repo.opensciencegrid.org/osg/3.4/osg-3.4-el7-release-latest.rpm \
+                   epel-release \
+                   yum-plugin-priorities
 
 RUN yum clean all && \
-    yum update -y && \
-    yum install -y frontier-squid && \
-    systemctl enable frontier-squid
+    yum update -y 
+
+RUN yum install -y frontier-squid && \
+    yum install -y supervisor
+
+RUN yum clean all --enablerepo=* && rm -rf /var/cache/yum/
+
 
-COPY app/ /app
+RUN systemctl enable frontier-squid
 
+ADD sbin/* /usr/local/sbin/
 
-EXPOSE 3128
```

Comment:
I'm not clear about how EXPOSE works in a k8s/helm world but can we not just let the image expose the (default ports 3128 and 3401) and let the orchestration layer handle the actual host ports?

Reply:
It is a no-op in k8s, as far as I understand. 
Not sure for helm.

Anyway, a container user could decide to configure the squid on a non-default port (for any reason), and in that case this command would be invalid.

BTW: We are not exposing anything for xcache, so this would be more in line with that anyway.

Reply:
I like this answer [here](https://stackoverflow.com/questions/22111060/what-is-the-difference-between-expose-and-publish-in-docker/47594352#47594352) where `EXPOSE` is mainly used as light documentation to tell the users which container ports they should be publishing. I vote that we keep it and think about doing this in the XCache images, if having a default port there is sensible.

Reply:
Not convinced it is a good idea, but can leave with it.